src/NotificationHandler.php

```diff
@@ -62,7 +66,26 @@ public function handle($username, $password, $xml)
         }
 
         $event = $this->parse($xml);
+        $message = $event->getMessage();
+        $response = $message->generateResponse();
+        $status = new Status();
+
+        $status->setType('request')
+          ->setTime(date('d/m/Y', time()))
+          ->setDate(date('h:i:s', time()));
+        if ($event->getName() == ExceptionEvent::NAME) {
+            $status->setCode('-1')
+              ->setMessage('NOK');
+        } else {
+            $status->setCode('0')
+              ->setMessage('OK');
+        }
+
         $this->eventDispatcher->dispatch($event->getName(), $event);
+
+        $poetry = new Poetry();
+
+        return $poetry->getRenderer()->render($response);
```

Comment:
We should inject dependencies as opposed to access them directly from the container.